src/field.ml

```diff
@@ -39,9 +45,11 @@ module MakeBool (Size : SIZE) = struct
   type t = bool [@@deriving repr]
 
   let size = Size.size
+  (* you should raise an error is size is not 1 *)
```

Comment:
Although I agree that it seems silly to have anything besides 1, we technically can waste bytes and encode booleans on more than 1 byte ?